cmd/config/main.go

```diff
@@ -236,123 +260,115 @@ var (
-
-	marketMap, err := createMarketMap()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "error generating the local market map")
-		return
-	}
+	// Set the update interval for the oracle.
+	LocalOracleConfig.UpdateInterval = *updateInterval
+	LocalOracleConfig.MaxPriceAge = *maxPriceAge
 
-	// If the tickers is not empty, filter the tickers to include only the tickers
-	// that are specified.
-	tickersFlag := *tickersOpt
-	if tickersFlag != "" {
-		for _, ticker := range marketMap.Tickers {
-			if !strings.Contains(tickersFlag, ticker.String()) {
-				delete(marketMap.Providers, ticker.String())
-				delete(marketMap.Paths, ticker.String())
-				delete(marketMap.Tickers, ticker.String())
-			}
-		}
+	if *debug {
+		LocalOracleConfig.Production = false
 	}
 
-	if len(providersFlag) > 0 {
-		for ticker, providers := range marketMap.Providers {
-			ps := make([]mmtypes.ProviderConfig, 0)
-
-			// Filter the providers to include only the providers that are specified.
-			for _, provider := range providers.Providers {
-				if strings.Contains(providersFlag, provider.Name) {
-					ps = append(ps, provider)
-				}
-			}
-
-			providers.Providers = ps
-			marketMap.Providers[ticker] = providers
-
-			// If there are no providers for a given ticker, remove the ticker from the
-			// market map.
-			if len(providers.Providers) == 0 {
-				delete(marketMap.Providers, ticker)
-				delete(marketMap.Paths, ticker)
-				delete(marketMap.Tickers, ticker)
-			}
-		}
+	if err := LocalOracleConfig.ValidateBasic(); err != nil {
+		fmt.Fprintf(os.Stderr, "error validating local config: %v\n", err)
+		return err
 	}
 
-	// Validate the market map.
-	if err := marketMap.ValidateBasic(); err != nil {
-		fmt.Fprintf(os.Stderr, "error validating the market map: %v\n", err)
-		return
+	f, err := os.Create(*oracleCfgPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error creating local config file: %v\n", err)
 	}
+	defer f.Close()
 
-	// Encode the local market config file.
-	encoder = json.NewEncoder(f)
+	// Encode the local oracle config file.
+	encoder := json.NewEncoder(f)
 	encoder.SetIndent("", "  ")
-	if err := encoder.Encode(marketMap); err != nil {
-		panic(err)
+	if err := encoder.Encode(LocalOracleConfig); err != nil {
+		return err
 	}
+
+	fmt.Fprintf(os.Stdout, "successfully created oracle config file at %s\n", *oracleCfgPath)
+	return nil
 }
 
 // createMarketMap creates a market map given all of the local market configurations for
 // each provider as well as the custom conversion markets. We do so to ensure that the
 // oracle is always started using the market map that is expected to be stored by the
 // market map module.
-func createMarketMap() (mmtypes.MarketMap, error) {
+func createMarketMap() error {
+	if *chain == dYdXChain {
+		fmt.Fprintf(
+			os.Stderr,
+			"dYdX chain requires the use of a predetermined market map. please use the market map provided by the Skip/dYdX team or the default market map provided in /config/dydx/market.json",
+		)
```

Comment:
dydx specific function.

if we have this all factored out already, it will be easier to reason about in the future